File: Construct_target_array_with_multiple_sums.cpp
class Solution {
public:
    bool isPossible(vector<int>& target){
        long long int sum = accumulate(target.begin(), target.end(), 0LL);
        priority_queue<int> pq (target.begin(), target.end());
        while(!pq.empty() and pq.top() > 1){
            int max_element = pq.top();
            pq.pop();
            sum -= max_element;
            if(max_element <= sum or sum < 1){
                return false;
            }
            int old_element = max_element%sum;
            sum += old_element;
            pq.push(old_element != 0 ? old_element : sum);
        }
        return true;
    }
};

File: Construct_target_array_with_multiple_sums.java
public class Solution {
    public boolean isPossible(int[] target) {
        long sum = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for (int num : target) {
            sum += num;
            pq.offer(num);
        }
        while (!pq.isEmpty() && pq.peek() > 1) {
            int maxElement = pq.poll();
            sum -= maxElement;
            if (maxElement <= sum || sum < 1) {
                return false;
            }
            int oldElement = (int)(maxElement % sum);
            sum += oldElement;
            pq.offer(oldElement != 0 ? oldElement : (int)sum);
        }
        return true;
    }
}
